tests/client.rs

```diff
@@ -23,35 +23,6 @@ async fn test_plain() {
     ClientBuilder::new(vec![connection]).build().await.unwrap();
 }
 
-#[tokio::test]
```

Comment:
I'm OK with removing this since `get_cached_leader` wasn't meant to be public anyways. On the long term it would probably nice to have two unit tests:

1. test that we can extract the partition leader from the Kafka metadata message
2. test that certain requests (produce and fetch IIRC) always use the partition leader and not some random broker